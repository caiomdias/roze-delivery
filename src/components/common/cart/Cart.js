import React, { useState } from 'react';
import ProductCard from '../productCard/ProductCard'
import Close from '../../../assets/images/svg/close';

const Cart = (props) => {
	const [isOpen, setOpen] = useState(true);

	return (
		<aside className={isOpen ? 'cart cart--open': 'cart'}>
			<div className="cart__header">
				<span className="title">Meu Carrinho</span>
				<Close
					className="close-btn"
					onClick={() => setOpen(false)}/>
			</div>
			<div className="cart__content">
				<ProductCard mode="cart" />
				<ProductCard mode="cart" />
				<ProductCard mode="cart" />
				<ProductCard mode="cart" />
				<ProductCard mode="cart" />
			</div>
			<div className="cart__total">
				<span>Total a pagar:</span>
				<strong className="total-amount">R$ 9,50</strong>
			</div>
		</aside>
	)
};

export default Cart;
