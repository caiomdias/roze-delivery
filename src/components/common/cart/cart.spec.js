import { italic } from 'chalk';
import { test } from 'picomatch';
// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Cart from './Cart';

describe("Cart Component", () => {
	it('render correctly Cart component', () => {  
		const cartComponent = renderer.create(<Cart />);
		expect(cartComponent).toMatchSnapshot();
	});
});