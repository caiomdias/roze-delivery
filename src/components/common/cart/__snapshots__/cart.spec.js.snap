// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component render correctly Cart component 1`] = `
<aside
  className="cart cart--open"
>
  <div
    className="cart__header"
  >
    <span
      className="title"
    >
      Meu Carrinho
    </span>
    <svg
      className="close-btn"
      enableBackground="new 0 0 413.348 413.348"
      onClick={[Function]}
      viewBox="0 0 413.348 413.348"
      width="30px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m413.348 24.354-24.354-24.354-182.32 182.32-182.32-182.32-24.354 24.354 182.32 182.32-182.32 182.32 24.354 24.354 182.32-182.32 182.32 182.32 24.354-24.354-182.32-182.32z"
      />
    </svg>
  </div>
  <div
    className="cart__content"
  >
    <div
      className="product-card product-card--cart"
    >
      <div
        className="product-card__content"
      >
        <div
          className="product-card__image"
        >
          <img
            alt="Imagem de undefined"
          />
        </div>
        <div
          className="product-card__description"
        >
          <h4
            className="product-card__title"
          />
          <p
            className="product-card__price"
          >
            R$ 
          </p>
        </div>
      </div>
      <div
        className="product-card__actions"
      >
        <div
          className="quantity"
        >
          <div
            className="quantity__minus"
            onClick={[Function]}
          >
            -
          </div>
          <div
            className="quantity__value"
          >
            0
          </div>
          <div
            className="quantity__plus"
            onClick={[Function]}
          >
            +
          </div>
        </div>
        <a
          href=""
        >
          Excluir
        </a>
      </div>
    </div>
    <div
      className="product-card product-card--cart"
    >
      <div
        className="product-card__content"
      >
        <div
          className="product-card__image"
        >
          <img
            alt="Imagem de undefined"
          />
        </div>
        <div
          className="product-card__description"
        >
          <h4
            className="product-card__title"
          />
          <p
            className="product-card__price"
          >
            R$ 
          </p>
        </div>
      </div>
      <div
        className="product-card__actions"
      >
        <div
          className="quantity"
        >
          <div
            className="quantity__minus"
            onClick={[Function]}
          >
            -
          </div>
          <div
            className="quantity__value"
          >
            0
          </div>
          <div
            className="quantity__plus"
            onClick={[Function]}
          >
            +
          </div>
        </div>
        <a
          href=""
        >
          Excluir
        </a>
      </div>
    </div>
    <div
      className="product-card product-card--cart"
    >
      <div
        className="product-card__content"
      >
        <div
          className="product-card__image"
        >
          <img
            alt="Imagem de undefined"
          />
        </div>
        <div
          className="product-card__description"
        >
          <h4
            className="product-card__title"
          />
          <p
            className="product-card__price"
          >
            R$ 
          </p>
        </div>
      </div>
      <div
        className="product-card__actions"
      >
        <div
          className="quantity"
        >
          <div
            className="quantity__minus"
            onClick={[Function]}
          >
            -
          </div>
          <div
            className="quantity__value"
          >
            0
          </div>
          <div
            className="quantity__plus"
            onClick={[Function]}
          >
            +
          </div>
        </div>
        <a
          href=""
        >
          Excluir
        </a>
      </div>
    </div>
    <div
      className="product-card product-card--cart"
    >
      <div
        className="product-card__content"
      >
        <div
          className="product-card__image"
        >
          <img
            alt="Imagem de undefined"
          />
        </div>
        <div
          className="product-card__description"
        >
          <h4
            className="product-card__title"
          />
          <p
            className="product-card__price"
          >
            R$ 
          </p>
        </div>
      </div>
      <div
        className="product-card__actions"
      >
        <div
          className="quantity"
        >
          <div
            className="quantity__minus"
            onClick={[Function]}
          >
            -
          </div>
          <div
            className="quantity__value"
          >
            0
          </div>
          <div
            className="quantity__plus"
            onClick={[Function]}
          >
            +
          </div>
        </div>
        <a
          href=""
        >
          Excluir
        </a>
      </div>
    </div>
    <div
      className="product-card product-card--cart"
    >
      <div
        className="product-card__content"
      >
        <div
          className="product-card__image"
        >
          <img
            alt="Imagem de undefined"
          />
        </div>
        <div
          className="product-card__description"
        >
          <h4
            className="product-card__title"
          />
          <p
            className="product-card__price"
          >
            R$ 
          </p>
        </div>
      </div>
      <div
        className="product-card__actions"
      >
        <div
          className="quantity"
        >
          <div
            className="quantity__minus"
            onClick={[Function]}
          >
            -
          </div>
          <div
            className="quantity__value"
          >
            0
          </div>
          <div
            className="quantity__plus"
            onClick={[Function]}
          >
            +
          </div>
        </div>
        <a
          href=""
        >
          Excluir
        </a>
      </div>
    </div>
  </div>
  <div
    className="cart__total"
  >
    <span>
      Total a pagar:
    </span>
    <strong
      className="total-amount"
    >
      R$ 9,50
    </strong>
  </div>
</aside>
`;
